docs/website/postprocess_dialect_docs.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# MLIR dialect markdown document postprocessor for website generation.
+
+import argparse
+import os
+import fileinput
+import re
+
+
+def main(args):
+    directory = args.directory
+    files = [os.path.join(directory, f) for f in os.listdir(directory)]
+
+    with fileinput.input(files=files, inplace=True) as f:
+        for line in f:
+            # Replace certain headings with one level deeper.
+            # Skipping heading levels is usually discouraged, but we aren't
+            # getting much value from treating these as subsections and we
+            # don't want them showing up in the rendered table of contents.
```

Comment:
When does max depth not already cover this?

Reply:
1. dialects with no groups: 
![image](https://github.com/openxla/iree/assets/4010439/c1732aa5-2c0a-42f9-baf1-b9ea4ed082cc)

2. attributes and types even when there are groups: 
![image](https://github.com/openxla/iree/assets/4010439/00854822-2812-4c5e-8aed-95cdcecbd216)

Reply:
Ah, OK, so you want to have the same max depth in the header specified across all files rather than one per and so making these groups at same nesting here.

Reply:
Per-file max depth would help with case 1 but not case 2 (parameters for attributes and types are the same depth regardless of if op definitions use groups or not). I'm also not quite sure if per-file max depth works without some edits to the site generator templates (could look into that again as needed).

Reply:
Added a comment with an example to explain how/why.